src/main/java/io/kokuwa/maven/helm/DependencyBuildMojo.java

```diff
@@ -28,6 +41,180 @@ public class DependencyBuildMojo extends AbstractHelmMojo {
+		throws MojoExecutionException {
+
+		//delete the outdated Chart.lock
+		File chartLock = new File(chartDirectory.toString() + "/Chart.lock");
+		
+		if (chartLock.exists()) {
+			chartLock.delete();
+			getLog().info("Chart.lock deleted at " + chartDirectory);
+		} else {
+			getLog().info("No Chart.lock file found at " + chartDirectory);
+		}
+
+		List<String> overwrittenChart = new ArrayList<>();
+		List<String> originalChart;
+		try {
+			originalChart = Files.readAllLines(Paths.get(chartFile.toURI()), StandardCharsets.UTF_8);
+		} catch (IOException e) {
+			throw new MojoExecutionException("Unable to read Chart.yaml at " + chartDirectory, e);
+		}
+
+		//Used for determining when a dependency has been read
+		boolean name, repository, version;
+		name = version = repository = false;
+
+		int currentDependencyIndex = 0;
+		Dependency currentDependency = dependencies.get(currentDependencyIndex);
+
+		for (String line : originalChart) {
+
+			//all dependencies have been read so just need to write remaining lines
+			if (currentDependencyIndex == dependencies.size()) {
+				overwrittenChart.add(line);
+				continue;
+			}
+
+			//Repository is not a required field so could be null
+			if (currentDependency.getRepository() == null) {
+				repository = true;
+			}
+
+			//check for lines that need to be overwritten
+			if (line.contains(" version:")) { 
+				version = true;
+				
+				if (currentDependency.isOverwrite()) {
+					getLog().info("Overwriting dependency '" + currentDependency.getName() + "' with new version:");
+					getLog().info("\tOld Value:");
+					getLog().info(line);
+
+					//preserve any characters/spacing that may appear before the attribute
+					int endIndex = line.indexOf(" version:") + "version:".length();
+					line = line.substring(0, endIndex + 1);
+					line += " " + overwriteDependencyVersion;
+
+					getLog().info("\tNew Value:");
+					getLog().info(line);
+				}
+			} else if (line.contains(" repository:")) {
+				repository = true;
+
+				if (currentDependency.isOverwrite()) {
+					getLog().info("Overwriting dependency '" + currentDependency.getName() + "' with new repository:");
+					getLog().info("\tOld Value:");
+					getLog().info(line);
+
+					//preserve any characters/spacing that may appear before the attribute
+					int endIndex = line.indexOf(" repository:") + "repository:".length();
+					line = line.substring(0, endIndex + 1);
+					line += " " + overwriteDependencyRepository;
+
+					getLog().info("\tNew Value:");
+					getLog().info(line);
+				}
+			} else if (line.contains(" name:")) { 
+					name = true;
+			}
+
+			overwrittenChart.add(line);
+
+			//check if the current dependency has been read completely
+			if (name && version && repository) {
+				currentDependencyIndex++;
+				name = version = repository = false;
+
+				if (currentDependencyIndex < dependencies.size()) {
+					currentDependency = dependencies.get(currentDependencyIndex);
```

Comment:
A: This is a bit of a stylistic choice so feel free to ignore, but an iterator might make this a bit cleaner